controllers/integration/integration_adapter.go

```diff
@@ -137,6 +137,139 @@ func (a *Adapter) getAllApplicationSnapshots() (*[]appstudioshared.ApplicationSn
 	return &applicationSnapshots.Items, nil
 }
 
+func (a *Adapter) getAllEnvironments() (*[]appstudioshared.Environment, error) {
+
+	environmentList := &appstudioshared.EnvironmentList{}
+	opts := []client.ListOption{
+		client.InNamespace(a.application.Namespace),
+	}
+	err := a.client.List(a.context, environmentList, opts...)
+	return &environmentList.Items, err
```

Comment:
I feel, to be safe, it is better to add code below before this line
```
	if err != nil {
		return nil, err
}
```

Reply:
I just notice this PR is on your own repo, I thought I didn't commit on the redhat-appstudio successful. So please ignore comment here.